packages/compass-e2e-tests/helpers/retry-with-backoff.ts

```diff
@@ -1,10 +1,12 @@
-const { delay } = require('./delay');
+import delay from './delay';
 
-async function retryWithBackoff(
-  fn = async () => {},
+export default async function retryWithBackoff(
+  fn = async () => {
+    return Promise.resolve();
+  },
   retries = 3,
   backoffStep = 200
-) {
+): Promise<any> {
```

Comment:
Something like this maybe:

```ts
export default async function retryWithBackoff<T>(
  fn: () => Promise<T>,
  retries = 3,
  backoffStep = 200
): ReturnType<typeof fn> {
```

In hindsight doesn't make sense to have a default value for `fn` anyway and allows to correctly type return type instead of `any`

Reply:
Cool. Lots of typescript I still have to learn. Or at least internalise.